core/include/scipp/core/element/sort.h

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Thibault Chatel
+#pragma once
+
+#include "scipp/common/overloaded.h"
+#include "scipp/common/span.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/transform_common.h"
+#include "scipp/core/value_and_variance.h"
+#include "scipp/units/unit.h"
+
+#include <stddef.h>
+
+namespace scipp::core::element {
+
+constexpr auto sort_common =
+    overloaded{core::element::arg_list<
+                   std::tuple<span<int64_t>, span<int32_t>, span<double>,
```

Comment:
I think you need to remove the `std::tuple` here (i.e., `arg_list<span<...>, ...>`), then https://github.com/scipp/scipp/pull/1288#discussion_r473747963 should work.